docs/src/main/asciidoc/security-authorization.adoc

```diff
@@ -0,0 +1,138 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Authorization of Web Endpoints
```

Comment:
@loicmathieu 
> Should the title be Authorization of Web Endpoints using configuration ? As securing Web Endpoints is also possible (partialy) using annotation ?

The plan is for this document to have all the information related to all types of the authorization, configuration and annotation based, and whatever else we may come up with, if it is not in this document then it will be added later I suppose. This PR is only about turning `security.adoc` into a front end `Security` page by pushing some of the current content into the new documents

Reply:
@loicmathieu Sorry, ignore the above, as I did not know the annotation based authorization was described in the other doc :-), I'll move it from there to here

Reply:
OK, this can be done later;

Reply:
OK, I understand it now, forget this comment ;)